packages/blade/src/components/Tooltip/Tooltip.web.tsx

```diff
@@ -0,0 +1,129 @@
+import { size } from '~tokens/global';
+
+const Tooltip = ({
+  content,
+  children,
+  placement = 'top',
+  onOpenChange,
+}: TooltipProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const id = useId();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const arrowRef = React.useRef<SVGSVGElement>(null);
+
+  const GAP = theme.spacing[2];
+  const [side] = getPlacementParts(placement);
+  const isHorizontal = side === 'left' || side === 'right';
+  const isOppositeAxis = side === 'right' || side === 'bottom';
+
+  const { refs, floatingStyles, context } = useFloating({
+    placement,
+    open: isOpen,
+    strategy: 'fixed',
+    onOpenChange: (open) => {
+      if (open) {
+        setIsOpen(true);
+        onOpenChange?.({ isOpen: open });
+      } else {
+        setIsOpen(false);
+        onOpenChange?.({ isOpen: open });
+      }
+    },
+    middleware: [
+      shift({ crossAxis: false, padding: GAP }),
+      flip({ padding: GAP }),
+      offset(GAP + ARROW_HEIGHT),
+      arrow({
+        element: arrowRef,
+        padding: isHorizontal ? 0 : ARROW_WIDTH,
+      }),
+    ],
+    whileElementsMounted: autoUpdate,
+  });
+
+  const animationOffset = isOppositeAxis ? -size[4] : size[4];
+  const { isMounted, styles } = useTransitionStyles(context, {
+    duration: theme.motion.duration.quick,
+    initial: {
+      opacity: 0,
+      transform: `translate${isHorizontal ? 'X' : 'Y'}(${animationOffset}px)`,
+    },
+  });
+
+  useDelayGroup(context, { id });
+  const { delay } = useDelayGroupContext();
+  const hover = useHover(context, {
+    delay,
+    move: false,
+  });
+  const focus = useFocus(context);
+  const role = useRole(context, { role: 'tooltip' });
+  const { getReferenceProps, getFloatingProps } = useInteractions([role, hover, focus]);
+
+  return (
+    <>
+      {/* Cloning the trigger children to enhance it with ref and event handler */}
+      {React.cloneElement(children, {
+        ref: refs.setReference,
+        ...getReferenceProps(),
+        ...makeAccessible({ label: content }),
+      })}
+      {isMounted && (
+        <FloatingPortal>
+          <BaseBox
+            ref={refs.setFloating}
+            style={floatingStyles}
+            pointerEvents="none"
+            {...getFloatingProps()}
+          >
+            <TooltipContent
+              style={styles}
+              arrow={
+                <FloatingArrow
+                  ref={arrowRef}
+                  context={context}
+                  width={ARROW_WIDTH}
+                  height={ARROW_HEIGHT}
+                  fill={theme.colors.brand.gray[200].highContrast}
+                  stroke={theme.colors.brand.gray[300].highContrast}
+                  strokeWidth={theme.border.width.thin}
+                />
+              }
+            >
+              {content}
+            </TooltipContent>
+          </BaseBox>
+        </FloatingPortal>
+      )}
+    </>
+  );
+};
+
+export { Tooltip, TooltipProps };
```

Comment:
do we need to export `TooltipProps` from here? lets do all the re-export from the `index.ts` these things sometimes lead to circular deps and we bang our heads as to where is this issue happening.

Reply:
Cool. done.